modules/library/src/main/scala/zio/elasticsearch/executor/HttpExecutor.scala

```diff
@@ -489,25 +497,36 @@ private[elasticsearch] final class HttpExecutor private (esConfig: ElasticConfig
             e => ZIO.fail(new ElasticException(s"Exception occurred: ${e.getMessage}")),
             value => ZIO.succeed((Chunk.fromIterable(value.results).map(Item.apply), value.scrollId))
           )
-        case _ =>
-          ZIO.fail(createElasticExceptionFromCustomResponse(response))
+        case errorCode =>
+          ZIO.fail(handleFailuresFromCustomResponse(errorCode, response))
       }
     }
 
-  private def createElasticException(response: Response[Either[String, String]]): ElasticException =
-    new ElasticException(
-      s"Unexpected response from Elasticsearch. Response body: ${response.body.fold(body => body, _ => "")}"
-    )
+  private def getQueryParams(parameters: List[(String, Any)]): ScalaMap[String, String] =
+    parameters.collect { case (name, Some(value)) => (name, value.toString) }.toMap
+
+  private def handleFailures(errorCode: StatusCode, response: Response[Either[String, String]]) =
+    errorCode match {
+      case HttpUnauthorized | HttpForbidden =>
+        UnauthorizedException
+      case _ =>
+        new ElasticException(
+          s"Unexpected response from Elasticsearch. Response body: ${response.body.fold(body => body, _ => "")}"
+        )
+    }
 
-  private def createElasticExceptionFromCustomResponse[A](
+  private def handleFailuresFromCustomResponse[A](
```

Comment:
Let's have only `response` here. So you can match `response.code` instead of `errorCode`.

Reply:
Makes sense!